pkg/windows/utils.go

```diff
@@ -31,9 +34,103 @@ var (
 	}
 )
 
+var softwareRegistryKey = "Software\\Rancher"
+var softwareCalicoRegistryKey = softwareRegistryKey+"\\Calico"
+
+
+func ensureRegistryKey() error {
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, "Software", registry.QUERY_VALUE)
+	if err != nil {
+		return err
+	}
+	defer k.Close()
+
+	softwareK, _, err := registry.CreateKey(k, "Rancher", registry.CREATE_SUB_KEY)
+	if err != nil {
+		return err
+	}
+	defer softwareK.Close()
+
+	calicoK, exist, err := registry.CreateKey(softwareK, "Calico", registry.CREATE_SUB_KEY)
+	if err != nil {
+		return err
+	}
+	calicoK.Close()
+	if !exist {
+		err = setStoredLastBootTime("")
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+//Get latest stored reboot
+func getStoredLastBootTime() (string, error) {
+	calicoK, err := registry.OpenKey(registry.LOCAL_MACHINE, softwareCalicoRegistryKey, registry.QUERY_VALUE)
+	if err != nil {
+		return "", err
+	}
+	defer calicoK.Close()
+
+	lastStoredBoot, _, err := calicoK.GetStringValue("LastBootTime")
+	if err != nil {
+		return "", err
+	}
+	return lastStoredBoot, nil
+}
+
+//Set last boot time on the registry
+func setStoredLastBootTime(lastBootTime string) error {
+	calicoK, err := registry.OpenKey(registry.LOCAL_MACHINE, softwareCalicoRegistryKey, registry.SET_VALUE)
+	if err != nil {
+		return err
+	}
+	defer calicoK.Close()
+
+	err = calicoK.SetStringValue("LastBootTime", lastBootTime)
+	if err != nil {
+		return err
+	}
+	
+	return nil
+}
+
+// Check if the node was rebooted
+func isNodeRebooted(ctx context.Context) (bool, error) {
+	path, err := exec.LookPath("powershell")
+	if err == nil {
+		cmd := exec.CommandContext(ctx, path, "-NoProfile", "-NonInteractive", "@(Get-CimInstance Win32_OperatingSystem).LastBootUpTime.ToString(\"G\")")
```

Comment:
You're still execing powershell. You should be able to do this using pure go. See for example: https://gistlib.com/go/get-system-boot-time-in-milliseconds-in-go

Reply:
I did something similar using a different function. Now I'm not using any exec command so I also removed the context and left as it was before.